### PR TITLE
Binding.implementation ts correction

### PR DESCRIPTION
**Pull request by tonyhallett**
## Description

This is being picky but for correctness.  For the generic T to be correct the implementationType type has to change otherwise T will be incorrect for cache/onActivation when toConstructor has been used to set the Binding.

## Motivation and Context

It corrects the typescript - e.g no need for the BindingToSyntax to cast to any when setting the implementationType in the toConstructor method.

## How Has This Been Tested?

npm run test on Windows machine

## Types of changes

- [ ] Updated docs / Refactor code / Added a tests case (non-breaking change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
Yes if using the Binding.implementationType.
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
No new tests required
- [x] All new and existing tests passed.
- [ ] I have updated the changelog.

### CODE REVIEW

**Review by notaphplover — APPROVED** [review]

File: src/bindings/binding.ts [hunk 1]
```diff
@@ -15,8 +15,8 @@ class Binding<T> implements interfaces.Binding<T> {
     // A runtime identifier because at runtime we don't have interfaces
     public serviceIdentifier: interfaces.ServiceIdentifier<T>;
 
-    // The constructor of a class which must implement T
-    public implementationType: interfaces.Newable<T> | null;
+    // The constructor of a class
+    public implementationType: interfaces.Newable<any> | null;
```
Comment (notaphplover) [inline src/bindings/binding.ts:19]: I would use `unknown` instead

File: src/interfaces/interfaces.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ namespace interfaces {
         dynamicValue: ((context: interfaces.Context) => T) | null;
         scope: BindingScope;
         type: BindingType;
-        implementationType: Newable<T> | null;
+        implementationType: Newable<any> | null;
```
Comment (notaphplover) [inline src/interfaces/interfaces.ts:54]: I would use `unknown` instead

**Review by notaphplover — CHANGES_REQUESTED** [review]
We are trying to avoid the use of any. As pointed previously, we shouldn't use the generic T in the implementationType property, but I think `unknown` is more appropiate even if additional casts may be added

**Comment by tonyhallett** [comment]
It is not just the implementationType that is incorrect.  Closing as this needs more consideration.

**Pull request closed without merging**